test/tests.py

```diff
@@ -0,0 +1,175 @@
+        - Lack of stage-1/MCprep_addon/ignore.blend
+        - "DEV MAIN" in mock_stdout
+        - "MAIN" in mock_stdout
+        - "hi guys c:" in stage-1/MCprep_addon/mcprep_dev.txt
+        """
+        with mock.patch(
+            "sys.argv",
+            ["bab", "-c", f"{TEST_FOLDER}/test_addon/bpy-build.yaml", "-b", "dev"],
+        ):
+            bab.main()
+        build = Path(f"{TEST_FOLDER}/test_addon/build")
+
+        # This could be consolidated into a single call,
+        # but I feel this is more readable as it's calling
+        # for each individual condition, and reduces complexity.
+        self.assertTrue(build.exists() and build.is_dir())
+        self.assertTrue((build / "MCprep_addon.zip").exists())
+        self.assertTrue((build / "stage-1").exists())
+        self.assertTrue((build / "stage-1/MCprep_addon/mcprep_dev.txt").exists())
+        self.assertFalse((build / "stage-1/MCprep_addon/ignore.blend").exists())
+
+        # Check mock_stdout and mcprep_dev.txt for some
+        # expected strings.
+        self.assertRegex(mock_stdout.getvalue(), r"DEV MAIN")  # dev action
+        self.assertRegex(mock_stdout.getvalue(), r"MAIN")  # default action
+
+        with open(build / "stage-1/MCprep_addon/mcprep_dev.txt", "r") as f:
+            self.assertEqual(f.read().strip(), "hi guys c:")
+
+    @mock.patch("sys.stdout", new_callable=StringIO)
+    def test_hooks(self, mock_stdout: StringIO) -> None:
+        """Perform a test build using the
+        project in test_addon.
+
+        This uses the standard default action, but
+        now checks for the output of different
+        hooks in mock_stdout.
+
+        To make sure this test is accurate, we
+        use the -s flag to suppress extra BpyBuild
+        output.
+
+        This test will check for:
+        - "PRE BUILD {TEST_FOLDER}/test/test_addon/MCprep_addon" in mock_stdout
+        - "MAIN {TEST_FOLDER}/test/test_addon/build/stage-1/MCprep_addon" in mock_stdout
+        - "POST INSTALL " in mock_stdout
+        - "CLEAN UP {TEST_FOLDER}/test/test_addon/MCprep_addon" in mock_stdout
+
+        TODO: add more complexity to hooks
+        """
+        with mock.patch(
+            "sys.argv",
+            ["bab", "-c", f"{TEST_FOLDER}/test_addon/bpy-build.yaml", "-s"],
+        ):
+            bab.main()
+
+        # Check mock_stdout for expected strings.
+        stdout_list = mock_stdout.getvalue().split("\n")
+        self.assertIn(f"PRE BUILD {TEST_FOLDER}/test_addon/MCprep_addon", stdout_list)
+        self.assertIn(
+            f"MAIN {TEST_FOLDER}/test_addon/build/stage-1/MCprep_addon", stdout_list
+        )
+        self.assertIn(f"CLEAN UP {TEST_FOLDER}/test_addon/MCprep_addon", stdout_list)
+
+        # TODO: Check folder of post install
+        #
+        # This is dependent on the system, so
+        # reliably checking this is much harder
+        self.assertRegex(mock_stdout.getvalue(), r"POST INSTALL ")
```

Comment:
We'll have to figure out a way to test for paths in `post_install`. The `post_install` hook in tests does output the path, but that path differs depending on the system and Blender versions found

Reply:
Resolved in [`205226d`](https://github.com/Moo-Ack-Productions/bpy-build/pull/9/commits/205226d56b95a5ac32e014d6aba7a8449860cfa6)